dataland-email-service/src/main/kotlin/org/dataland/datalandemailservice/services/EmailSubscriptionManager.kt

```diff
@@ -0,0 +1,82 @@
+package org.dataland.datalandemailservice.services
+
+import org.dataland.datalandemailservice.email.EmailSender
+import org.dataland.datalandemailservice.repositories.EmailSubscriptionRepository
+import org.dataland.datalandmessagequeueutils.messages.InternalEmailMessage
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import java.util.UUID
+
+/**
+ * Service responsible for managing email subscriptions.
+ */
+@Service
+class EmailSubscriptionManager(
+    @Autowired private val emailSubscriptionRepository: EmailSubscriptionRepository,
+    @Autowired private val emailSender: EmailSender,
+    @Autowired private val internalEmailBuilder: InternalEmailBuilder,
+) {
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    /**
+     * Unsubscribes an email address from email communications based on the provided UUID and sends a mail
+     * to the stakeholders that this email address is now unsubscribed.
+     * @param uuid The UUID of the subscription to be unsubscribed.
+     * @return A response entity indicating the result of the unsubscription.
+     */
+    @Transactional
+    fun unsubscribeUuidAndInformStakeholders(uuid: UUID): ResponseEntity<String> {
+        val emailSubscription = emailSubscriptionRepository.findByUuid(uuid)
+
+        return if (emailSubscription != null) {
+            unsubscribeUuid(emailSubscription.uuid)
+            informStakeholdersOfUnsubscription(emailSubscription.emailAddress)
+            val successMessage = "Successfully unsubscribed email address corresponding to UUID: $uuid."
+            logger.info(successMessage)
+            ResponseEntity.ok(successMessage)
+        } else {
+            val errorMessage = "There is no email address corresponding to UUID: $uuid."
+            logger.info(errorMessage)
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(errorMessage)
+        }
+    }
+
+    /**
+     * Unsubscribes an email based on the provided UUID.
+     *
+     * This method sets the `isSubscribed` flag to `false` for the email subscription entity
+     * identified by the given UUID. If no entity is found with the provided UUID, the method
+     * performs no action.
+     *
+     * @param uuid The UUID of the email subscription to unsubscribe.
+     * @return This method has no return value.
+     *
+     */
+    private fun unsubscribeUuid(uuid: UUID) {
```

Comment:
I think, this function can be put into the one above, ie no necessity to declare it as an individual function. It's just two lines of code and not reused.

Reply:
done